qutip/core/cy/coefficient.pyx

```diff
@@ -12,30 +13,69 @@ cimport cython
 import qutip
 
 cdef extern from "<complex>" namespace "std" nogil:
-    double complex cos(double complex x)
     double complex conj(double complex x)
     double         norm(double complex x)
-    double complex exp(double complex x)
+
 
 cdef class Coefficient:
-    cdef double complex _call(self, double t) except *:
-        return 0j
+    """
+    `Coefficient` are the time-dependant scalar of a `[Qobj, coeff]` pair
+    composing time-dependant operator in list format for :obj:`QobjEvo`.
 
-    cpdef void arguments(self, dict args) except *:
-        self.args.update(args)
+    :obj:`Coefficient` are immutable.
+    """
+    def __init__(self):
+        raise NotImplementedError("Only sub-classes should be initiated.")
+
+    def replace_arguments(self, _args=None, **kwargs):
```

Comment:
We can't change it until Python 3.8 is our lowest supported version, but an interesting note: in recent versions, we can actually do this pattern without even preventing the use of a variable name:
```python
>>> def f(x, /, **kwargs):
...     return x, kwargs
>>> f(1)
(1, {})
>>> f(x=1)
(None, {'x': 1})
>>> f(1, x=2):
(1, {'x': 2})
```
Of course it would be a logical error for a user to actually do that in this case, but still an interesting corner of the language!

Reply:
(edited post to correct the signature...)